File: LeetCode/Study Plan/Practice Problems/Hard/Data Stream As Disjoint Intervals/Go/dataStreamAsDisjointIntervalsBruteForce.go
//below is the brute force solution

type SummaryRanges struct {
	intervals map[int]bool
}

func Constructor() SummaryRanges {
	return SummaryRanges{intervals: make(map[int]bool)}
}

func (this *SummaryRanges) AddNum(value int) {
	this.intervals[value] = true
}

func (this *SummaryRanges) GetIntervals() [][]int {
	l, r := -1, -1
	res := [][]int{}
	for i := 0; i < int(1e4); i++ {
		if this.intervals[i] {
			if l == -1 {
				l, r = i, i
			} else {
				r = i
			}
		} else {
			if l != -1 && r != -1 {
				res = append(res, []int{l, r})
			}
			l, r = -1, -1
		}
	}
	return res
}

/**
 * Your SummaryRanges object will be instantiated and called as such:
 * obj := Constructor();
 * obj.AddNum(value);
 * param_2 := obj.GetIntervals();
 */

File: LeetCode/Study Plan/Practice Problems/Hard/Data Stream As Disjoint Intervals/Java/dataStreamAsDisjointIntervalsBruteForce.java
class SummaryRanges {
    private Set<Integer> values;
    public SummaryRanges() {
        values = new TreeSet<>();
    }
    
    public void addNum(int value) {
        values.add(value);
    }
    
    public int[][] getIntervals() {
        if (values.isEmpty()) {
            return new int[0][2];
        }
        List<int[]> intervals = new ArrayList<>();
        int left = -1;
        int right = -1;
        for (Integer value : values) {
            if (left == -1) {
                left = right = value;
            } else if ((right+1) == value) {
                right = value;
            }else {
                intervals.add(new int[]{left, right});
                left = right = value;
            }
        }
        intervals.add(new int[]{left, right});
        return intervals.toArray(new int[0][]);
    }
}

/**
 * Your SummaryRanges object will be instantiated and called as such:
 * SummaryRanges obj = new SummaryRanges();
 * obj.addNum(value);
 * int[][] param_2 = obj.getIntervals();
 */
